FluentAssertions.Core/Types/TypeAssertions.cs

```diff
@@ -198,6 +200,366 @@ public AndConstraint<TypeAssertions> BeDecoratedWith<TAttribute>(string because
+        /// <param name="reasonArgs">Zero or more objects to format using the placeholders in <see cref="!:because"/>.</param>
+        public AndConstraint<TypeAssertions> Implement(Type interfaceType, string because = "", params object[] reasonArgs)
+        {
+            if (!interfaceType.IsInterface)
+            {
+                throw new ArgumentException("Must be an interface Type.", "interfaceType");
+            }
+
+            Execute.Assertion.ForCondition(Subject.GetInterfaces().Contains(interfaceType))
+                .BecauseOf(because, reasonArgs)
+                .FailWith("Expected type {0} to implement interface {1}{reason}, but it does not.", Subject, interfaceType);
+
+            return new AndConstraint<TypeAssertions>(this);
+        }
+
+        /// <summary>
+        /// Asserts that the current <see cref="Type"/> does not implement Interface <paramref name="interfaceType"/>.
+        /// </summary>
+        /// <param name="interfaceType">The interface that should be implemented.</param>
+        /// <param name="because">A formatted phrase as is supported by <see cref="M:System.String.Format(System.String,System.Object[])"/> explaining why the assertion
+        ///             is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.</param>
+        /// <param name="reasonArgs">Zero or more objects to format using the placeholders in <see cref="!:because"/>.</param>
+        public AndConstraint<TypeAssertions> NotImplement(Type interfaceType, string because = "", params object[] reasonArgs)
+        {
+            if (!interfaceType.IsInterface)
+            {
+                throw new ArgumentException("Must be an interface Type.", "interfaceType");
+            }
+
+            Execute.Assertion.ForCondition(!Subject.GetInterfaces().Contains(interfaceType))
+                .BecauseOf(because, reasonArgs)
+                .FailWith("Expected type {0} to not implement interface {1}{reason}, but it does.", Subject, interfaceType);
+
+            return new AndConstraint<TypeAssertions>(this);
+        }
+
+        /// <summary>
+        /// Asserts that the current <see cref="Type"/> is derived from <see cref="Type"/> <paramref name="baseType"/>.
+        /// </summary>
+        /// <param name="baseType">The Type that should be derived from.</param>
+        /// <param name="because">A formatted phrase as is supported by <see cref="M:System.String.Format(System.String,System.Object[])"/> explaining why the assertion
+        ///             is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.</param>
+        /// <param name="reasonArgs">Zero or more objects to format using the placeholders in <see cref="!:because"/>.</param>
+        public AndConstraint<TypeAssertions> BeDerivedFrom(Type baseType, string because = "", params object[] reasonArgs)
```

Comment:
I'd like to get some additional opinions here. I chose "Be" because it implies some permanence. DeriveFrom felt a bit more like an ongoing activity. That said, Implements makes sense as-is and doesn't require [cannot handle?] the Be preamble, so maybe trimming it is right.